Content.Server/Nyanotrasen/Abilities/Psionics/Abilities/TelegnosisPowerSystem.cs

```diff
@@ -48,6 +48,10 @@ private void OnShutdown(EntityUid uid, TelegnosisPowerComponent component, Compo
 
         private void OnPowerUsed(EntityUid uid, TelegnosisPowerComponent component, TelegnosisPowerActionEvent args)
         {
+            // This is technically a sort of mind swap targeted on onesself, so even item granted ones will not work if you have insulation.
+            if (HasComp<PsionicInsulationComponent>(uid))
+                return;
```

Comment:
Shouldn't there be an API for this already. There's PsionicInsulation, PsionicDisabled, etc.